### PR TITLE
logging non-existent projects/groups in forProjects/forGroups

### LINKED ISSUE #1572: groups/projects' existence should checked when installing authorization stack
Stack with non-existent entries in `forGroups`/`forProjects` should be accepted however project/group existence should be checked and properly logged to (WARN).

### PR DESCRIPTION
**Pull request by tulinkry**
fixes #1572

### CODE REVIEW

**Comment by vladak** [comment]
The AuthorizationPluginTest.java looks identical to what is currently in AuthorizationStackTest.java. Test case for invalid project/group would be nice.

File: src/org/opensolaris/opengrok/authorization/AuthorizationEntity.java [hunk 1]
```diff
@@ -369,25 +375,47 @@ public void setForGroups(String[] groups) {
...
         Set<String> groups = new TreeSet<>();
+
         for (String x : forGroups()) {
             /**
              * Full group discovery takes place here. All projects/repositories
              * in the group are added into "forProjects" and all subgroups
              * (including projects/repositories) and parent groups (excluding
              * the projects/repositories) are added into "forGroups".
+             *
+             * If the group does not exist then an warning is issued.
```
Comment (vladak) [inline src/org/opensolaris/opengrok/authorization/AuthorizationEntity.java]: a

Comment (tulinkry) [inline src/org/opensolaris/opengrok/authorization/AuthorizationEntity.java]: fixed

**Comment by tulinkry** [comment]
> The AuthorizationPluginTest.java looks identical to what is currently in AuthorizationStackTest.java. Test case for invalid project/group would be nice.

I deduplicated the code, now testing both plugin and stack for the same.
I added a test case for invalid project and group.

**Pull request merged**